Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resource/ProblemResource.tsx

```diff
@@ -21,35 +21,37 @@ const ProblemResource = ({ resource }: IProblemResourceProps) => {
     const {
         actions: {
             getProblemResourceFile,
-            getProblemResourceResponse,
+            problemResourceContent,
+            clearFileContentState,
         },
     } = useHomeContests();
 
     const saveFile = useCallback(() => {
-        if (!getProblemResourceResponse) {
+        if (!problemResourceContent) {
             return;
         }
 
         // todo: move this to http helper
-        const filename = getProblemResourceResponse
+        const filename = problemResourceContent
             .headers['content-disposition']
             .split('filename*=UTF-8\'\'')[1];
 
         const filenameDecoded = decodeURIComponent(filename);
```

Comment:
Why is this not in the `hook`?

This can become just `saveProblemResouce()` and all the logic will be moved inside the hook

Reply:
moved this in the http hook